Xamarin.Essentials/Calendar/Calendar.android.cs

```diff
@@ -80,7 +85,41 @@ static async Task<IEnumerable<DeviceEvent>> PlatformGetEventsAsync(string calend
                         EndDate = cur.GetInt(eventsProjection.IndexOf(CalendarContract.Events.InterfaceConsts.AllDay)) == 0 ? (DateTimeOffset?)DateTimeOffset.FromUnixTimeMilliseconds(cur.GetLong(eventsProjection.IndexOf(CalendarContract.Events.InterfaceConsts.Dtend))) : null
                     });
                 }
-                return events;
+            }
+            if (events.Count == 0)
+            {
+                // Make sure this calendar exists by testing retrieval
+                GetCalendarById(calendarId);
```

Comment:
Don't rely on solely on the method throwing an exception:
1) Wrap this in a try catch. If exception is ArgumentOutOfRange or ArgumentException it can be re-thrown. If other exception, then wrap in an ArgumentException
2) Check the return value isn't null - if null, raise ArgumentException